hangupsbot/plugins/cleverbot/__init__.py

```diff
@@ -0,0 +1,77 @@
+# Set your API key by saying
+# /bot config set cleverbot-api-key "EXAMPLE_API_KEY_REPLACE_THIS"
+# to your bot
+
+from cleverwrap import CleverWrap
+from random import randrange, randint
+import asyncio
+import plugins
+import logging
+
+logger = logging.getLogger(__name__)
+__cleverbots = dict()
+
+def _initialise(bot):
+    plugins.register_handler(_handle_incoming_message, type="message")
+    plugins.register_user_command(["chat"])
+    plugins.register_admin_command(["chatreset"])
+
+@asyncio.coroutine
+def _handle_incoming_message(bot, event, command):
+    """Handle random message intercepting"""
+
+    if not event.text:
+        return
+
+    if not bot.get_config_suboption(event.conv_id, 'cleverbot_percentage_replies'):
+        return
+
+    percentage = bot.get_config_suboption(event.conv_id, 'cleverbot_percentage_replies')
+
+    if randrange(0, 101, 1) < float(percentage):
+        yield from chat(bot, event)
+
+def _get_cw_for_chat(bot, event):
+    segregate = bot.get_config_suboption(event.conv_id, "cleverbot_segregate")
```

Comment:
segregate should be turned on by default, as this replicates the original behaviour of the plugin